### ISSUE TITLE
grouped apply and transform w/ an agg seems to be inconsistent with docs

### ISSUE THREAD

**Issue by machow** [dask/dask#5982]
Hey--really enjoying using dask, and tutorial.dask.org is super helpful!

The groupby apply and transform docs contain a warning that agg operations will be run within each grouping x partition pair. (shown below)

![image](https://user-images.githubusercontent.com/2574498/76045490-3c697c80-5f2b-11ea-9f8a-89332c38ca9b.png)

This behavior makes a lot of sense. However, looking through issues, it doesn't seem like the behavior described in the warning is happening. The aggs seem to only happen within each grouping (as in the pandas groupby). In light of this, I was wondering...

* is there a clear case where the behavior from the warning occurs?
* the warning is phrased in way that sounds like applying to each grouping x partition pair will always happen. If the examples below make sense, and this isn't always the case--could it be helpful to clarify that in the warning?

I could be totally wrong in my examples, but here are two below...

### Example 1

https://github.com/dask/dask/issues/4526 used this example:

```python
import dask.dataframe as dd
import pandas as pd
df = pd.DataFrame([dict(x='x{}'.format(i % 3), y='y{}'.format(i)) for i in range(9)])
ddf = dd.from_pandas(df, 4)

meta = pd.Series([], name="y", index=pd.Index([], name="x"), dtype=str)
ddf.groupby('x')['y'].apply('-'.join, meta=meta).compute()
```

output:

```python
x
x0    y0-y6-y3
x1    y1-y7-y4
x2    y2-y8-y5
Name: y, dtype: object
```

Here, one result was returned per grouping (3 levels of x), rather than for each of the 4 partitions.

### Example 2

```python
import pandas as pd
df = pd.DataFrame({
    "g": ['a', 'a', 'a', 'b', 'b', 'b'],
    "x": [1,2,3,4,5,6]
})

ddf = dd.from_pandas(df, 4)

f = lambda x: x.values.mean()

ddf.groupby('g').x.transform(f).compute()
df.groupby('g').x.transform(f)
```

Both output:

```python
0    2
1    2
2    2
4    5
5    5
3    5
Name: x, dtype: int64
```

**Comment by mrocklin** [comment]
>   aggs seem to only happen within each grouping (as in the pandas groupby)

Yes.  This is correct.  Dask will re-arrange all rows so that each group lives in only one partition.  It sounds like the warning is misleading, and so should maybe be rephrased.

**Comment by pavithraes** [comment]
Resolved by #8507
